students/vvinodh/Lesson9/mailroom_hw9.py

```diff
@@ -0,0 +1,134 @@
+from operator import itemgetter
+import os
+import sys
+import datetime
+
+
+class Donor:
+    def __init__(self, donation=None):
+        self.donation = donation if donation else []
+        self.total = sum(self.donation)
+        self.ave = self.total/len(self.donation) if donation else 0
+        self.num_donations = len(self.donation)
+        self.last = self.donation[-1]
+
+donor_data = {
+        'William Gates, III': [65000.00, 2, 326892.24],
+        'Mark Zuckerberg': [16000.00, 3, 5465.37],
+        'Jeff Bezos': [877.33, 1, 877.33],
+        'Paul Allen': [708.42, 3, 236.14],
+              }
+
+
+class Donor_ops:
+
+
+    def donor_append(name, don):
+        try:
+            donor_data[name].append(don)
+        except KeyError:
+            donor_data.setdefault((name), []).append(don)
+
+    def thank_you(query):
+        if query != 'list' and query != 'menu':
+            donation_info = Donor(donor_data[query])
+            thank_you_text = str(f"""\n---------------- Thank You ----------------\n
+Dear {query},\n
+Thank you for your contribution of ${donation_info.last:,.2f}.
+It will be put to very good use.\n
+Sincerely,
+Vinodh""")
+
+            return thank_you_text
+        elif query == 'list':
+            return Donor_ops.list_donors()
+        else:
+            return 'returning to main menu...'
+
+
+    def list_donors():
+        list_text = str("\n---------------- List of Donors ----------------\n\n")
+        for donor in donor_data:
+            list_text += (donor + '\n')
+        return list_text
+
+    def report():
+        columns = ["DONOR", "TOTAL", "AVERAGE", "NUMBER"]
+        report_text = str("\n--------------- Report -------------\n\n")
+        report_text += str('{:<30}{:<15}{:<15}{:<15}\n'.format(*columns))
+        for name in donor_data:
+            donation_info = Donor(donor_data[name])
```

Comment:
instead of creating a Donor object every time you need to use, this class should hold reference to collection of these objects, for example
```
donor_collection = DonorCollection(Donor("jeff", [100, 200]), Donor("bob", [139]))
```
then functions like report can reference collection of donors that was initialized at the time of creating of donor collection object.